snooty/rstparser.py

```diff
@@ -944,14 +944,19 @@ def make_toc_entry(
         title: Optional[str] = None
         url: Optional[str] = None
         slug: Optional[str] = None
+        ref_project: Optional[str] = None
         if match:
             title, target = match["label"], match["target"]
+            # pipelines denote project reference
+            if target.startswith("|") and target.endswith("|"):
+                ref_project = target[1:-1]
+                target = None
         else:
             target = child
 
         if not title and util.PAT_URI.match(target):
```

Comment:
I think we'll want to add an `or` clause here for the second part of this conditional, since the `PAT_URI` regex of `^(?P<schema>[a-z]+)://` should be matching on `://`, which we won't have in the reference.

Maybe just `and (util.PAT_URI.match(target) or ref_project):`?
the `target.startswith("|") and target.endswith("|")` is also suitable as a boolean, if we want to assign to a var and reuse.

Reply:
oh good catch! added the ref_project case for erroneous node